### ISSUE TITLE
Feature problem: Cyclic __proto__ value

### ISSUE THREAD

**Issue by Yi2255** [googleprojectzero/fuzzilli#452]
### Problem
Code generators that use `setPrototype` may result in a cyclic assignment of `__proto__` value.

### Example
`const o10 = {`
`    __proto__: 64,`
`};`
`const o11 = {`
`    __proto__: o10,`
`};`
`o10.__proto__ = o11;`  //TypeError: Cyclic `__proto__` value

### The problem is more likely to be reproduced with the following generators weighting.
[cyclic assignment.txt](https://github.com/user-attachments/files/17359867/cyclic.assignment.txt)

**Comment by saelo** [comment]
Right, this can happen. I'm not sure how big of an issue this is, but what we could do is create a dedicated `SetPrototype` IL operation and always use [guards](https://github.com/googleprojectzero/fuzzilli/blob/0e20cd55a3f81997243716cf286bd7265238e5e8/Docs/HowFuzzilliWorks.md?plain=1#L78) for it so that we emit a try-catch in JavaScript.

**Comment by Yi2255** [comment]
I understand your point.  Thank you for sharing this idea!
